dbt_platform_helper/providers/schema_migrator.py

```diff
@@ -0,0 +1,80 @@
+from collections import Counter
+from collections import OrderedDict
+from copy import deepcopy
+from typing import Protocol
+
+from dbt_platform_helper.platform_exception import PlatformException
+from dbt_platform_helper.providers.io import ClickIOProvider
+from dbt_platform_helper.providers.schema_migrations.schema_v0_to_v1_migration import (
+    SchemaV0ToV1Migration,
+)
+from dbt_platform_helper.providers.version import InstalledVersionProvider
+
+
+class InvalidMigrationConfigurationException(PlatformException):
+    pass
+
+
+class PlatformConfigSchemaMigrationException(PlatformException):
+    pass
+
+
+class SchemaMigrationProtocol(Protocol):
+    def from_version(self) -> int: ...
+
+    def migrate(self, platform_config: dict) -> dict: ...
+
+
+ALL_MIGRATIONS = [SchemaV0ToV1Migration()]
```

Comment:
Now they are in a subdirectory, we could do this programatically.

I would accept a todo suggesting this for the next iteration though.